cmake/kokkos_arch.cmake

```diff
@@ -51,7 +51,8 @@ KOKKOS_ARCH_OPTION(PASCAL61        GPU  "NVIDIA Pascal generation CC 6.1")
 KOKKOS_ARCH_OPTION(VOLTA70         GPU  "NVIDIA Volta generation CC 7.0")
 KOKKOS_ARCH_OPTION(VOLTA72         GPU  "NVIDIA Volta generation CC 7.2")
 KOKKOS_ARCH_OPTION(TURING75        GPU  "NVIDIA Turing generation CC 7.5")
-KOKKOS_ARCH_OPTION(EPYC            HOST "AMD Epyc architecture")
```

Comment:
I feel like we need to ask ourselves: "Is it OK to just drop the option?"

Should we look for it and stop generation if it is provided?

Or is the warning issue by CMake good enough?
```
CMake Warning:
  Manually-specified variables were not used by the project:

    Kokkos_ARCH_EPYC
```

Reply:
Perhaps something to the effect of a `KOKKOS_DEPRECATED_ARCHITECTURES` list, if a user selects something from that list we give a message? I could get behind that idea

Reply:
We decided this would be a CMake error

Reply:
Pushed new commit with proposed error checking for removed options.